src/core_codemods/sonar_numpy_nan_equality.py

```diff
@@ -0,0 +1,42 @@
+import libcst as cst
+from codemodder.codemods.base_codemod import Reference
+from codemodder.codemods.libcst_transformer import (
+    LibcstTransformerPipeline,
+)
+from codemodder.codemods.sonar import SonarDetector
+from codemodder.codemods.sonar import SonarCodemod
+
+from core_codemods.api import Metadata
+from core_codemods.numpy_nan_equality import (
+    NumpyNanEquality,
+    NumpyNanEqualityTransformer,
+)
+
+
+class SonarNumpyNanEqualityTransformer(NumpyNanEqualityTransformer):
+    def leave_Comparison(
+        self, original_node: cst.Comparison, updated_node: cst.Comparison
+    ) -> cst.BaseExpression:
+        if self.filter_by_result(self.node_position(original_node)):
```

Comment:
Maybe I'm missing something but shouldn't this filter just be moved to the parent class?

Reply:
The parent class has an empty detector. This means that `self.results` is empty. Both `filter_by_result` and `node_is_selected` would fail in that instance.

Reply:
Makes sense, thanks. Although it also would be useful if we could just define `filter_by_result` to be vacuously true in those cases.